src/fides/api/api/v1/endpoints/generic_overrides.py

```diff
@@ -85,5 +96,92 @@ async def list_dataset_paginated(
     return await async_paginate(db, filtered_query, pagination_params)
 
 
+async def create_with_key(
+    data: Union[DataUse, DataCategory, DataSubject],
+    model: Type[Union[DataUse, DataCategory, DataSubject]],
+    db: AsyncSession,
+) -> Dict:
+    """
+    Helper to create taxonomy resource when not given a fides_key.
+    Automatically re-enables disabled resources with the same name.
+    """
+    # If data with same name exists but is disabled, re-enable it
+    disabled_resource_with_name = db.query(model).filter(
+        model.key == data.name,  # type: ignore[union-attr]
+        model.active is False,
+    )
+    if disabled_resource_with_name:
+        return model.update(db=db, data=disabled_resource_with_name, active=True)  # type: ignore[union-attr]
```

Comment:
What happens if the incoming `data` doesn't match the DB model we're reactivating? Should we pass in the incoming `data` instead of the `disabled_resource_with_name`?
```suggestion
        return model.update(db=db, data=data, active=True)  # type: ignore[union-attr]
```

Reply:
yes, good catch!